compiler/record-minmax/driver/Driver.cpp

```diff
@@ -114,17 +124,35 @@ int entry(const int argc, char **argv)
 
   RecordMinMax rmm(num_threads);
 
+  // TODO: support parallel record for profile with random data
+  if (num_threads > 1 and not arser["--input_data"])
+  {
+    throw std::runtime_error("Parallel recording is used only for h5 now\n");
+  }
+
   // Initialize interpreter and observer
   rmm.initialize(input_model_path);
 
   if (arser["--input_data"])
   {
     auto input_data_path = arser.get<std::string>("--input_data");
 
+    // TODO: support parallel record from file and dir input data format
+    if (num_threads > 1 and not(input_data_format == "h5") and not(input_data_format == "hdf5"))
+    {
+      throw std::runtime_error("Parallel recording is used only for h5 now\n");
+    }
+
     if (input_data_format == "h5" || input_data_format == "hdf5")
     {
       // Profile min/max while executing the H5 data
-      rmm.profileData(mode, input_data_path, min_percentile, max_percentile);
+      if (num_threads == 1)
+        rmm.profileData(mode, input_data_path, min_percentile, max_percentile);
+      else
+      {
+        std::cout << "Using parallel recording" << std::endl;
```

Comment:
there's `add_verbose()` used. maybe show this message only when some level is given.